lib/mcm/search/advanced_search.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module MCM
+  module Search
+    # Advanced seach functionalities
+    module Advanced
+      module_function
+
+      def search(term, _mechanism)
+        all = find_all
+
+        criteria = []
+        criteria.push(find_peroxy) if term[:peroxy]
+
+        criteria.reduce(all) { |first, second| first.intersect(second) }
+      end
+
+      def find_peroxy
+        valid_smarts = ['CO[O]', '[O]OC']
+
+        DB.fetch(
+          'SELECT * FROM Species WHERE Smiles NOT NULL AND (substruct_match(Smiles, ?) OR substruct_match(Smiles, ?))',
```

Comment:
Try keeping literal SQL to a minimum, so here only the `substract_match` portion should use it.
Something like 
```
DB[:Species]
  .exclude(.... remove null smiles, see comment below)
  .where(Sequel.lit("substruct_match(Smiles, ?) OR .....")
```